server/src/game/world.ts

```diff
@@ -218,60 +196,68 @@ export class World {
   }
 
   // map, civs
-  updateCivTileVisibility(civID: number): void {
-    const cityTiles: Coords[] = [];
-    this.map.forEachTile((tile, coords) => {
-      tile.clearVisibility(civID);
-      if (tile.owner?.civID === civID) {
-        tile.setVisibility(civID, true);
-        cityTiles.push(coords)
-      }
+  updateLeaderTileVisibility(leader: Leader): void {
+    leader.forEachCityDomainID((domainID: CityDomainID) => {
+      this.updateCityTileVisibility(domainID);
+    });
+
+    leader.forEachCivDomainID((domainID: CivDomainID) => {
+      this.updateCivTileVisibility(domainID);
     });
+  }
+
+  updateCityTileVisibility(domainID: CityDomainID): void {
+    const city = this.getCity(domainID);
+    const cityTiles: Coords[] = [];
+    for (const coords of city.getTiles()) {
+      this.map.getTileOrThrow(coords).setVisibility(domainID, true);
+      cityTiles.push(coords);
+    }
     for (const coords of cityTiles) {
       for (const neighbor of this.map.getNeighborsCoords(coords, 1, { filter: (tile) => {
-        return tile.owner?.civID !== civID;
+        return tile.owner?.id !== city.id;
       } })) {
         const tile = this.map.getTileOrThrow(neighbor);
-        tile.setVisibility(civID, true);
+        tile.setVisibility(domainID, true);
       }
     }
-    const civ = this.civs[civID];
-    for (const unit of civ.units) {
+
+    for (const unit of city.units) {
       for (const coords of this.map.getVisibleTilesCoords(unit)) {
         const tile = this.map.getTileOrThrow(coords);
-        tile.setVisibility(civID, true);
+        tile.setVisibility(domainID, true);
       }
     }
   }
 
-  // civs
-  getCivUnits(civID: number): Unit[] {
-    return this.civs[civID].getUnits();
-  }
-
-  // civs
-  getCivUnitPositions(civID: number): Coords[] {
-    return this.civs[civID].getUnitPositions();
+  updateCivTileVisibility(domainID: CivDomainID): void {
```

Comment:
Shouldn't this also loop through cities?

Reply:
You might be right about that... I'll have to test what the code actually currently does.